control_panel_api/tests/test_views.py

```diff
@@ -54,6 +76,22 @@ def test_detail(self):
         self.assertIn('users3buckets', response.data)
         self.assertEqual(8, len(response.data))
 
+        userapp = response.data['userapps'][0]
+        self.assertIn('id', userapp)
+        self.assertIn('app', userapp)
+        self.assertIn('is_admin', userapp)
+        self.assertEqual(3, len(userapp))
+
+        self.assert_app(userapp['app'])
+
+        users3bucket = response.data['users3buckets'][0]
+        self.assertIn('id', users3bucket)
+        self.assertIn('s3bucket', users3bucket)
+        self.assertIn('access_level', users3bucket)
+        self.assertEqual(4, len(users3bucket))
+
+        self.assert_s3bucket(users3bucket['s3bucket'])
```

Comment:
I wonder if this `assert_s3bucket` could get the `s3bucket` from the response and the expected s3bucket and check they much ... or may be too much?

Reply:
Yeh bit much, I'm just checking keys and have refactored common code.